src/AbstractRepository.php

```diff
@@ -39,6 +57,22 @@ protected function validateSearchLanguage($language)
     }
 
     /**
+     * Validate a path in contained in the base directory of this repository.
+     *
+     * @param string $path A file or directory path.
+     *
+     * @throws InvalidArgumentException If the given path in not in the base directory.
+     */
+    protected function validatePathInBaseDirectory($path)
+    {
+        if (!Path::isBasePath($this->baseDir, $path)) {
+            throw new InvalidArgumentException(
+                sprintf('Resource "%s" is not in the repository base directory "%s"', $path, $this->baseDir)
+            );
```

Comment:
I usually format exceptions with `sprintf()`s like this to make them easier to read:

```php
throw new InvalidArgumentException(sprintf(
    'Resource "%s" is not in the repository base directory "%s"', 
    $path, 
    $this->baseDir
));
```

Could you change this?